src/init.cpp

```diff
@@ -132,6 +162,21 @@ void updateConfiguration(hpx::program_options::variables_map const& vm, configur
                            "num-np-cuda-streams-per-thread");
   updateConfigurationValue(vm, cfg.num_hp_cuda_streams_per_thread, "NUM_HP_CUDA_STREAMS_PER_THREAD",
                            "num-hp-cuda-streams-per-thread");
+
+  std::string mpi_mech_str = "";
+  updateConfigurationValue(vm, mpi_mech_str, "MPI_MECH", "mpi-mech");
+  if (mpi_mech_str == "yielding") {
+    cfg.mpi_mech = comm::MPIMech::Yielding;
+  }
+  else if (mpi_mech_str == "polling") {
+    cfg.mpi_mech = comm::MPIMech::Polling;
+  }
+  else if (mpi_mech_str != "") {
+    std::cout << "Unknown value for --mech=" << mpi_mech_str << "!" << std::endl;
+    std::terminate();
+  }
```

Comment:
Could this be moved into `convertFromString` instead?

Reply:
Yes, I refactored this part of the code. (I had to introduce another function but it's cleaner)